### ISSUE TITLE
Creating second migration fails in Postgres

### ISSUE THREAD

**Issue by chrisbrantley** [mikro-orm/mikro-orm#1341]
**Describe the bug**
Creating an initial migration works fine, but calling the `yarn migration:create` command to create subsequent migrations fails with an error.

**Stack trace**
```
error: select relname as constraint_name, attname as column_name, idx.indisunique as unique, idx.indisprimary as primary
      from pg_index idx
      left join pg_class AS i on i.oid = idx.indexrelid
      left join pg_attribute a on a.attrelid = idx.indrelid and a.attnum = ANY(idx.indkey) and a.attnum > 0
      where indrelid = '"null"."api_key"'::regclass - schema "null" does not exist
    at Parser.parseErrorMessage (/Users/chrisbrantley/code/enerflo-next/node_modules/pg-protocol/src/parser.ts:357:11)
    at Parser.handlePacket (/Users/chrisbrantley/code/enerflo-next/node_modules/pg-protocol/src/parser.ts:186:21)
    at Parser.parse (/Users/chrisbrantley/code/enerflo-next/node_modules/pg-protocol/src/parser.ts:101:30)
    at Socket.<anonymous> (/Users/chrisbrantley/code/enerflo-next/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:486:12)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:284:9)
    at Socket.Readable.push (_stream_readable.js:223:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23) {
  length: 99,
  severity: 'ERROR',
  code: '3F000',
  detail: undefined,
  hint: undefined,
  position: '326',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'namespace.c',
  line: '3040',
  routine: 'get_namespace_oid'
}
```

**To Reproduce**
1. Create an initial migration with `migration:create`
2. Apply the migration with `migration:up`
3. Attempt to create a new migration with `migration:create`

**Expected behavior**
I would expect the new migration to be created without error.

**Additional context**
I'm using the postgresql driver

**Versions**

| Dependency | Version |
| - | - |
| node | v14.25.0 |
| typescript | v4.0.2 |
| mikro-orm | v4.4.0 |
| your-driver | v4.4.0 |

**Comment by B4nan** [comment]
I can assure you this wont reproduce anything. My bet is you have the postgres misconfigured, but it is almost certainly something with its setup.

**Comment by chrisbrantley** [comment]
I appreciate the response. When you say "My bet is you have the postgres misconfigured". Do you mean the Postgres server itself or do you mean the PostgreSqlDriver options in the Mikro-Orm config?

**Comment by chrisbrantley** [comment]
@B4nan I think I found the problem:

If you look [here](https://github.com/mikro-orm/mikro-orm/blob/9875a9490cdb7a03edd35c7178a8a610f6f1a5a5/packages/postgresql/src/PostgreSqlSchemaHelper.ts#L68) in PostgreSqlSchemaHelper.ts you can see there is `nullif(table_schema, 'public') as schema_name`. So the schema will be `null` for the vast majority of cases where the default "public" schema is used.

That value gets passed to `getIndexesSql()` right [here](https://github.com/mikro-orm/mikro-orm/blob/9875a9490cdb7a03edd35c7178a8a610f6f1a5a5/packages/postgresql/src/PostgreSqlSchemaHelper.ts#L170). That method has a default of "public" for `schemaName` but since the passed in value is `null` (and not `undefined`) the default is not being applied. And since `null` is not a valid schema name the query fails.

This doesn't happen on the initial migration because the database schema is empty so `getIndexesSql()` is never called. But as soon as there is a table in the database this error will happen.

**Comment by chrisbrantley** [comment]
Looking closer it looks like I updated most packages to `4.4.0` but I missed `@mikro-orm/cli` which was still on `4.2.0` which caused an older version of `@mikro-orm/knex` to be installed. When I resolved that and ensured I was running the latest version of `@mikro-orm/knex` the issue is resolved because the schema name is set to `undefined` via the nullish coalescing operator here: https://github.com/mikro-orm/mikro-orm/blob/9875a9490cdb7a03edd35c7178a8a610f6f1a5a5/packages/knex/src/schema/DatabaseSchema.ts#L12

Now that the versions are all corrected it is working as expected. Closing.
